apis/v1alpha1/gateway_types.go

```diff
@@ -398,6 +398,39 @@ type RouteBindingSelector struct {
 	Resource string `json:"resource"`
 }
 
+// RouteNamespaces indicate which namespaces Routes should be selected from.
+type RouteNamespaces struct {
+	// NamespaceSelector is a selector of namespaces that Routes should be
+	// selected from. This is a standard Kubernetes LabelSelector, a label query
+	// over a set of resources. The result of matchLabels and matchExpressions
+	// are ANDed. Controllers must not support Routes in namespaces outside this
+	// selector.
+	//
+	// An empty selector (default) indicates that Routes in any namespace can be
+	// selected.
+	//
+	// The OnlySameNamespace field takes precedence over this field. This
+	// selector will only take effect when OnlySameNamespace is false.
+	//
+	// Support: Core
+	//
+	// +optional
+	NamespaceSelector metav1.LabelSelector `json:"namespaceSelector"`
```

Comment:
Should this be named `namespace`?
We have had feedback that we shouldn't have `selector` in the name.

Reply:
I think this may be out of scope for this PR since this is just moving the fields from GatewayClass to Gateway types files. I do agree the name may be a bit too repetitive (`routes.routeNamespaces.namespaceSelector`). I would actually lean the other way though - that we might want to change the name to `routes.namespaces.selector`. Maybe we need a follow up issue for this?

Reply:
https://github.com/kubernetes-sigs/service-apis/issues/311